### ISSUE TITLE
Replace input/output geometry type stubs

### ISSUE THREAD

**Issue by viddo** [CartoDB/cartodb#7053]
When the point-in-polygon type was implemented (#7031) the definitions of the [input/output geometry types of analysis nodes were hardcoded](https://github.com/CartoDB/cartodb/blob/b2a40b60d5133954ac9cdf959aba6ef83fbee3d2/lib/assets/javascripts/cartodb3/data/camshaft-reference.js#L74-L110) in the local camshaft reference wrapper (per prior agreement offline w/ rochoa since he was going on vacation at the time). The valid-input/output geometry types should be defined in the [camshaft reference](https://github.com/CartoDB/camshaft/blob/master/reference/versions) instead.

TBD:
- [ ] add definitions in reference; need to define format/requirements?
- [ ] add `operation`param for point-in-polygon to mandate wanted output geometry of point-in-polygon?
- [ ] enough to use the (current) simplified geometry type or need the original one? (e.g. `point` vs ST_Point`/`ST_MultiPoint`/…?
- [ ] replace hardcoded stub implementation

**Comment by viddo** [comment]
ping @rochoa @javierarce @xavijam

**Comment by xavijam** [comment]
:+1:

**Comment by javierarce** [comment]
:ok_hand:

**Comment by rochoa** [comment]
About
>  - [ ] add definitions in reference; need to define format/requirements?

Here is my initial proposal: https://github.com/CartoDB/camshaft/pull/5/files

**Comment by rochoa** [comment]
About

> - [ ] add `operation` param for point-in-polygon to mandate wanted output geometry of point-in-polygon?

I've just realised it's not that easy to make that to the reference as it is something that has to be evaluated, probably the best option is to have two different node types in those cases so it's easier to serialise.

I did a proof of concept for the library itself, as per https://github.com/CartoDB/camshaft/commit/8af31804e6418967c566c82584f10733b774abf9, but that takes advantage of having full access to node context. The reference cannot access the library but only a JSON file.
